lib/public/views/SimulationPasses/PerLhcPeriodOverview/SimulationPassesPerLhcPeriodOverviewModel.js

```diff
@@ -0,0 +1,152 @@
+import { SortModel } from '../../../components/common/table/SortModel.js';
+import { TextTokensFilterModel } from '../../../components/Filters/common/filters/TextTokensFilterModel.js';
+import { OverviewPageModel } from '../../../models/OverviewModel.js';
+import { RemoteData } from '/js/src/index.js';
+import { ObservableData } from '../../../utilities/ObservableData.js';
+import { getRemoteData } from '../../../utilities/fetch/getRemoteData.js';
+
+/**
+ * Simulation Passes Per LHC Period overview model
+ */
+export class SimulationPassesPerLhcPeriodOverviewModel extends OverviewPageModel {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super();
+        this._overviewSortModel = new SortModel();
+        this._overviewSortModel.observe(() => {
+            this._pagination.silentlySetCurrentPage(1);
+            this.load();
+        });
+        this._overviewSortModel.visualChange$.bubbleTo(this);
+
+        this._namesFilterModel = new TextTokensFilterModel();
+        this._registerFilter(this._namesFilterModel);
+
+        this._lhcPeriod = new ObservableData(RemoteData.notAsked());
+        this._lhcPeriod.bubbleTo(this);
+
+        this._lhcPeriodId = null;
+    }
+
+    /**
+     * Fetch LHC Period data which simulation passes are fetched
+     * @return {Promise<void>} promise
+     */
+    async _fetchLhcPeriod() {
+        this._lhcPeriod.setCurrent(RemoteData.loading());
+        try {
+            const { data: lhcPeriodStatistics } = await getRemoteData(`/api/lhcPeriodsStatistics/${this._lhcPeriodId}`);
+            this._lhcPeriod.setCurrent(RemoteData.success({ ...lhcPeriodStatistics, ...lhcPeriodStatistics.lhcPeriod }));
+        } catch (error) {
+            this._lhcPeriod.setCurrent(RemoteData.failure(error));
+        }
+    }
+
+    // eslint-disable-next-line valid-jsdoc
+    /**
+     * @inheritdoc
+     */
+    async load() {
+        this._fetchLhcPeriod();
+        super.load();
+    }
+
+    // eslint-disable-next-line valid-jsdoc
+    /**
+     * @inheritdoc
+     */
+    getRootEndpoint() {
+        const params = {
+            filter: {
+                names: this._namesFilterModel.normalized,
+                lhcPeriodIds: [this._lhcPeriodId],
+            },
+        };
+
+        const { appliedOn: sortOn, appliedDirection: sortDirection } = this._overviewSortModel;
+        if (sortOn && sortDirection) {
+            params[`sort[${sortOn}]`] = sortDirection;
+        }
+
+        return buildUrl('/api/simulationPasses', params);
+    }
+
+    /**
+     * Reset this model to its default
+     *
+     * @returns {void}
+     */
+    reset() {
+        this._namesFilterModel.reset();
+        super.reset();
+    }
+
+    /**
+     * Set id of LHC Period which runs are to be fetched
+     * @param {number} lhcPeriodId id of Data Pass
+     */
+    set lhcPeriodId(lhcPeriodId) {
+        this._lhcPeriodId = lhcPeriodId;
+    }
```

Comment:
No reload of data?

Reply:
no, it is called separately, as for RunsPerDataPass

Reply:
Ok